src/ExpressionEvaluator/Core/Source/ExpressionCompiler/DkmUtilities.cs

```diff
@@ -323,6 +323,8 @@ internal MetadataContextItem(TMetadataContext metadataContext)
             {
                 this.MetadataContext = metadataContext;
             }
+
+            protected override void OnClose() { }
```

Comment:
Please add a comment that this override is needed to keep the `DkmDataItem` from being unloaded ahead of this component. Perhaps include a link to #22617.

Reply:
Thank you!

---
In reply to: [143836754](https://github.com/dotnet/roslyn/pull/22644#discussion_r143836754) [](ancestors = 143836754)